projects/core/abstract/abstract-driver.directive.ts

```diff
@@ -1,20 +1,28 @@
-import {Directive, Inject, Self} from '@angular/core';
+import {Directive, Inject, OnInit, Self} from '@angular/core';
 import {TuiDestroyService} from '@taiga-ui/cdk';
-import {Observable} from 'rxjs';
+import {merge, Observable} from 'rxjs';
 import {distinctUntilChanged, takeUntil} from 'rxjs/operators';
 
 import {TuiDriver} from './driver';
 import {TuiVehicle} from './vehicle';
 
 @Directive()
-export abstract class AbstractTuiDriverDirective {
+export abstract class AbstractTuiDriverDirective implements OnInit {
+    abstract type: string;
```

Comment:
does it need to be unified?

```ts
abstract type: 'dropdown' | 'hint' | 'viewport';
```

Reply:
It's a general approach, can be anything really in future.